Tusk 2-1/Задание 2-1.cpp

```diff
@@ -0,0 +1,83 @@
+#define _USE_MATH_DEFINES 
+#include <iostream>
+#include <iomanip>
+#include <cmath>
+
+
+/**
+* \brief Calculation of the perimeter of a right triangle
+ * \param First Cathet - Length of the first cathet
+ * \param Second Cathet - Length of the second cathet
+ * \return The perimeter of a right triangle
+ */
+double getPerimeter(double FirstCathet, double SecondCathet);
+
+/**
+ * \brief Calculation of the area of a right triangle
+ * \param First Cathet - Length of the first cathet
+ * \param Second Cathet - Length of the second cathet
+ * \return The area of a right triangle
+ */
+double getArea(double FirstCathet, double SecondCathet);
+
+/**
+ *\brief User choice of perimeter calculation (1) or area (0)
+ */
+enum class State { perimeter, area };
+
+/**
+* \brief Program entry point
+* \return Error code (0) success
+*/
+int main()
+{
+
+	double FirstCathet, SecondCathet;
+	std::cout << "Length of the first cathet - ";
+	std::cin >> FirstCathet;
+	std::cout << "Length of the second cathet - ";
+	std::cin >> SecondCathet;
+	std::cout << "Select an action (to calculate the area, select 0, to calculate the perimeter, select 1)";
+	int input;
+	std::cin >> input;
+	const auto choice = static_cast<State>(input);
+
+
+	switch (choice)
+	{
+	case State::perimeter:
+	{
+		const double Perimeter = getPerimeter(FirstCathet, SecondCathet);
+
+		std::cout << "The perimeter of the triangle = " << Perimeter;
+		break;
+	}
+	case State::area:
+	{
+		const double Area = getArea(FirstCathet, SecondCathet);
+		std::cout << "The area of the triangle = " << Area;
+		break;
+	}
+	}
+	return 0;
+}
+
+
+double getPerimeter(double FirstKatet, double SecondKatet)
```

Comment:
Если параметры функции не изменяются, то они явно должны быть обозначены `const`
[By Alex Allain](https://www.cprogramming.com/tutorial/const_correctness.html)
>The const keyword allows you to specify whether or not a variable is modifiable. You can use const to prevent modifications to variables and const pointers and const references prevent changing the data pointed to (or referenced).
>Const дает вам возможность более четко документировать вашу программу и фактически обеспечивать соблюдение этой документации. Ключевое слово const обеспечивает вашим пользователям гарантии, позволяющие оптимизировать производительность без угрозы повреждения их данных, обеспечивая соблюдение вашей документации. Например, константные ссылки позволяют указать, что упомянутые данные не будут изменены; это означает, что вы можете использовать константные ссылки как простой и немедленный способ повышения производительности для любой функции, которая в настоящее время принимает объекты по значению, не беспокоясь о том, что ваша функция может изменить данные. Даже если это произойдет, компилятор предотвратит компиляцию кода и предупредит вас о проблеме. С другой стороны, если бы вы не использовали константные ссылки, у вас не было бы простого способа гарантировать, что ваши данные не были изменены.